### PR TITLE
Clean up unneeded method

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Review by ramaraochavali — APPROVED** [review]

**Comment by ramaraochavali** [comment]
@hzxuzhonghu It seems @nmittler changed them on purpose https://github.com/istio/istio/pull/38112#discussion_r835318097.

**Review by zirain — APPROVED** [review]
lgtm for E&T part.

**Comment by hzxuzhonghu** [comment]
@ramaraochavali No needless allocation here.

**Review by esnible — APPROVED** [review]
istioctl change looks fine.

File: pkg/util/sets/string.go [hunk 1]
```diff
@@ -18,43 +18,28 @@ import "sort"
...
-func NewWith(items ...string) Set {
-	return NewWithLength(len(items)).InsertAll(items...)
-}
-
-// Insert adds the item to the set.
-func (s Set) Insert(item string) Set {
-	s[item] = struct{}{}
-	return s
+// New creates a new Set with the given items.
+func New(items ...string) Set {
```
Comment (howardjohn) [inline pkg/util/sets/string.go:28]: I think the concern was `items ...string` always allocates an array?

Comment (hzxuzhonghu) [reply pkg/util/sets/string.go:28]: It doesnot matter, if no args, it is an empty slice. No mem gc at all.

**Review by howardjohn — APPROVED** [review]

File: pkg/util/sets/string.go (hunk 1 above)
Comment (nmittler) [reply pkg/util/sets/string.go:28]: It's probably fine. If there is any additional overhead it will be minimal.

**Comment by nmittler** [comment]
Created https://github.com/istio/istio/pull/38204 to revert `Set.InsertAll`.

**Pull request merged**